#! /bin/sh /usr/share/dpatch/dpatch-run
## 20glibc_has_grantpt.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: glibc has grantpt(), so only hceck for HAVE_SYS5_PTY
## DP: on non-glibc installations.

@DPATCH@
diff -Naur nvi-1.81.6.orig/ex/ex_script.c nvi-1.81.6/ex/ex_script.c
--- nvi-1.81.6.orig/ex/ex_script.c	2007-11-18 17:41:42.000000000 +0100
+++ nvi-1.81.6/ex/ex_script.c	2008-05-01 18:24:06.000000000 +0200
@@ -23,7 +23,8 @@
 #include <sys/select.h>
 #endif
 #include <sys/stat.h>
-#ifdef HAVE_SYS5_PTY
+/* glibc2.1 defines grantpt but there is no stropts.h */
+#if defined(HAVE_SYS5_PTY) && !defined(__GLIBC__)  
 #include <sys/stropts.h>
 #endif
 #include <sys/time.h>
@@ -664,7 +665,7 @@
 	F_CLR(gp, G_SCRWIN);
 }
 
-#ifdef HAVE_SYS5_PTY
+#if defined(HAVE_SYS5_PTY) && !defined(__GLIBC__)
 static int ptys_open __P((int, char *));
 static int ptym_open __P((char *));
 
